pkg/infrastructure/services/deployer_resources.go

```diff
@@ -55,34 +55,37 @@ func (s *serviceDeployer) createRequiredResources() (resources map[reflect.Type]
 		service := createRequiredService(s.instance, deployment)
 
 		appProps := map[string]string{}
-		var envProperties []corev1.EnvVar
+		var infraVolumes []v1alpha1.KogitoInfraVolume
 
 		if len(s.instance.GetSpec().GetInfra()) > 0 {
 			log.Debugf("Infra references are provided")
-			infraAppProps, infraEnvProp, err := s.fetchKogitoInfraProperties()
+			var infraAppProps map[string]string
+			var infraEnvProp []corev1.EnvVar
+			infraAppProps, infraEnvProp, infraVolumes, err = s.fetchKogitoInfraProperties()
 			if err != nil {
 				return resources, err
 			}
 			util.AppendToStringMap(infraAppProps, appProps)
-			envProperties = append(envProperties, infraEnvProp...)
+			deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env, infraEnvProp...)
 		}
 
 		if len(s.instance.GetSpec().GetConfig()) > 0 {
 			log.Debugf("custom app properties are provided")
 			util.AppendToStringMap(s.instance.GetSpec().GetConfig(), appProps)
 		}
 
-		s.applyEnvironmentPropertiesConfiguration(envProperties, deployment)
-
 		contentHash, configMap, err := getAppPropConfigMapContentHash(s.instance, appProps, s.client)
 		if err != nil {
 			return resources, err
 		}
-		s.applyApplicationPropertiesConfigurations(contentHash, deployment, s.instance)
+
+		s.applyApplicationPropertiesAnnotations(contentHash, deployment)
+
+		s.mountVolumes(infraVolumes, deployment)
```

Comment:
all the volume supported types are supported both by kubernetes and ocp?

Reply:
Yes, they are. It's part of the core API.